### PR TITLE
Fix multi-asic behaviour for mmuconfig

### PR DESCRIPTION
**Pull request by bktsim-arista**
#### What I did
Previously, mmuconfig did not function correctly on multi-asic devices as it did not traverse through the correct namespaces, as required for multi-asic devices. This change adds multi-asic support to mmuconfig with the use of multi-asic helper libraries, ensuring that mmuconfig searches throuugh the correct namespaces when used on multi-asic devices, and adds the '-n' optional argument for users to specify namespaces on multi-asic devices.

This is a part of the set of changes being pushed for https://github.com/sonic-net/sonic-buildimage/issues/15148

#### How I did it
Added namespace option -n and used multi_asic library to implement multi_asic behaviour. Added relevant unit tests to ensure functionality.

#### How to verify it
Run unit tests, or mmuconfig with `-n` option.

### CODE REVIEW

**Comment by vmittal-msft** [comment]
@bktsim-arista please re-base to latest

**Comment by vmittal-msft** [comment]
@bktsim-arista  please share command and it's output from multi-asic platform. Also, did we verify to make sure, it still works for ToR.

**Review by vmittal-msft — COMMENTED** [review]
Please check above comments.

**Comment by rdjeric-arista** [comment]
@vmittal-msft 
Here's the output on a multiasic platform:
```root@cmp235-3:~# mmuconfig -l
No buffer pool information available
No buffer profile information available
root@cmp235-3:~# mmuconfig -l -n asic0
No buffer pool information available
No buffer profile information available
root@cmp235-3:~# mmuconfig -p alpha_profile -a 2 -n asic0
root@cmp235-3:~# mmuconfig -l
No buffer pool information available
No buffer profile information available
root@cmp235-3:~# mmuconfig -l -n asic0
No buffer pool information available
Profile: alpha_profile
----------  -
dynamic_th  2
----------  -

root@cmp235-3:~#```
Also, we tested this on a ToR, and behavior was consistent on a single ASIC system.

File: tests/mmuconfig_input/mmuconfig_test_vectors.py [hunk 1]
```diff
@@ -83,6 +83,36 @@
...
 
+show_mmu_config_one = """\
+Pool: ingress_lossy_pool
+----  -------
+mode  dynamic
+type  ingress
+----  -------
+
+Pool: ingress_lossless_pool_hbm
+----  ---------
```
Comment (vmittal-msft) [inline tests/mmuconfig_input/mmuconfig_test_vectors.py:94]: There is no hbm pool in voq chassis. Can you please have example showing something like this ?

image

Comment (kenneth-arista) [reply tests/mmuconfig_input/mmuconfig_test_vectors.py:94]: @vmittal-msft the unit tests uses canned data in the mock_tables. The CLI show/config commands operate on CONFIG_DB and does not involve hardware programming. As such, it should be irrelevant whether or not the pool is HBM or not. Here's a code reference to the mock_tables: https://github.com/sonic-net/sonic-utilities/blob/d1ca905e3b0733170d19abeecad1cfbfd0180698/tests/mock_tables/config_db.json#L1958

**Comment by kenneth-arista** [comment]
@vmittal-msft @wenyiz2021 @arlakshm this PR is ready for review

**Comment by vmittal-msft** [comment]
@bktsim-arista  Please rebase

**Comment by pandyah5** [comment]
> @bktsim-arista  Please rebase

@bktsim-arista is not at Arista anymore. You can redirect these requests to me or @kenneth-arista. Thank you!

**Comment by arista-hpandya** [comment]
> @bktsim-arista Please rebase

@bktsim-arista is not at Arista anymore. You can redirect these requests to me or @kenneth-arista. Thank you!

**Review by vmittal-msft — APPROVED** [review]

**Comment by arista-hpandya** [comment]
@vmittal-msft Can we merge this change?

**Comment by bingwang-ms** [comment]
@bktsim-arista Please raise another PR to 202405 branch to address conflict.

**Pull request merged**